androidClient/app/src/main/java/com/macbitsgoa/comrades/HomeActivity.java

```diff
@@ -46,20 +51,41 @@
     private FragmentManager fragmentManager = getSupportFragmentManager();
     private SearchView searchView;
     private MySimpleDraweeView userProfileImage;
+    private FloatingActionButton fab_add_course;
+    public static CoordinatorLayout snack;
+    @SuppressLint("RestrictedApi")
     private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
             = item -> {
         switch (item.getItemId()) {
             case R.id.navigation_home:
+                fab_add_course.setVisibility(View.GONE);
+                fab_add_course.setOnClickListener(null);
+                Fragment homeFragment=fragmentManager.findFragmentByTag("HomeFragment");
+                if(homeFragment==null){
+                    homeFragment=HomeFragment.newInstance();
+                }
                 fragmentManager.beginTransaction().replace(R.id.container_fragment,
-                        HomeFragment.newInstance()).commit();
+                        homeFragment,"HomeFragment").commit();
                 return true;
             case R.id.navigation_courses:
+                fab_add_course.setVisibility(View.VISIBLE);
+                fab_add_course.setOnClickListener(v -> CourseListFragment.handleAddCourse(HomeActivity.this));
+                Fragment courseListFragment=fragmentManager.findFragmentByTag("CourseListFragment");
+                if(courseListFragment==null){
+                    courseListFragment= CourseListFragment.newInstance();
+                }
                 fragmentManager.beginTransaction().replace(R.id.container_fragment,
-                        CourseListFragment.newInstance()).commit();
+                        courseListFragment,"CourseListFragment").addToBackStack(null).commit();
                 return true;
             case R.id.navigation_profile:
+                fab_add_course.setVisibility(View.GONE);
+                fab_add_course.setOnClickListener(null);
+                Fragment profileFragment=fragmentManager.findFragmentByTag("ProfileFragment");
+                if(profileFragment==null){
+                    profileFragment=ProfileFragment.newInstance();
```

Comment:
Delete these `.newInstance()` as they are plain constructor calls and nothing more.